### PR TITLE
Corr 2: Add 'corr' aggregator to Python

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
stacked on #4491

File: hail/python/hail/expr/aggregators/aggregators.py [hunk 1]
```diff
@@ -1325,6 +1325,34 @@ def _linreg(y, x, nested_dim):
 
     return _agg_func('LinearRegression', _to_agg(y), t, [k, k0], seq_op_args=[lambda y: y, x])
 
+@typecheck(x=expr_float64, y=expr_float64)
+def corr(x, y) -> Float64Expression:
+    """Computes the Pearson correlation coefficient between `x` and `y`.
```
Comment (jbloom22) [inline hail/python/hail/expr/aggregators/aggregators.py]: wiki link to https://en.wikipedia.org/wiki/Pearson_correlation_coefficient

File: hail/python/test/hail/expr/test_expr.py [hunk 2]
```diff
@@ -512,6 +513,14 @@ def test_aggregator_group_by_sorts_result(self):
...
         self.assertEqual(grouped_expr, hl.eval(hl.sorted(grouped_expr)))
 
+    def test_agg_corr(self):
+        ht = hl.utils.range_table(10)
+        ht = ht.annotate(x=hl.rand_unif(-10, 10),
+                         y=hl.rand_unif(-10, 10))
+        c, xs, ys = ht.aggregate((hl.agg.corr(ht.x, ht.y), hl.agg.collect(ht.x), hl.agg.collect(ht.y)))
+
+        scipy_corr, _ = pearsonr(xs, ys)
+        self.assertAlmostEqual(c, scipy_corr)
```
Comment (jbloom22) [inline hail/python/test/hail/expr/test_expr.py]: add a small test case with missingness

Comment (tpoterba) [reply hail/python/test/hail/expr/test_expr.py]: I test missingness in scala

**Pull request merged**